### PR TITLE
blog: add Qos-class resources blog post

### PR DESCRIPTION
**Pull request by marquiz**
Create a blog post for introducing the [QoS-class resources KEP](https://github.com/kubernetes/enhancements/pull/3004).

### CODE REVIEW

**Comment by marquiz** [comment]
ping @kad @haircommander @SergeyKanzhelev @dchen1107 @mikebrow @mrunalp

File: content/en/blog/_posts/qos-class-resources.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+It is important to think about the future steps to see the proposal in a
+broader context and understand what the full solution would look like.
+
+- Pod Spec: replace pod annotations with proper user interface in the PodSpec.
+- Resource discovery: a mechanism for the runtime to communicate available
+  QoS-class resource types and the available classes within each resource type
+  to kubelet
+- Resource status/capacity: hold the information of discovered resources (e.g.
+  in node status) e.g. for users to see what is available
```
Comment (chrisnegus) [inline content/en/blog/_posts/qos-class-resources.md:119]:
```suggestion
  in node status) for users to see what is available
```

**Review by chrisnegus — COMMENTED** [review]
This reads very well, I have a few small editing suggestions.

**Comment by Sea-n** [comment]
Hey @marquiz,

It's been a while since the last update, would you like to address issues in the [previous comments](https://github.com/kubernetes/website/pull/34880#pullrequestreview-1042328871) so that we can move forward?

**Comment by marquiz** [comment]
> It's been a while since the last update, would you like to address issues in the [previous comments](https://github.com/kubernetes/website/pull/34880#pullrequestreview-1042328871) so that we can move forward?

Hi! Thanks all for the reviews and apologies for the silence. I just returned from a long summer vacation 🙄 

Yes, I want to address the issues and will look into those later this week.

**Comment by sftim** [comment]
To reduce the risk of somebody merging this to the wrong site, I'm also going to mark it as draft.

File: content/en/blog/_posts/qos-class-resources.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+traditionally been out of control in a Kubernetes environment, such as cache
+allocation, memory, and I/O bandwidth. This blog post introduces the concept of
+QoS-class resources, talks about the plan to integrate them into Kubernetes,
+and describes what is already possible with the latest container runtimes.
+
+## What are QoS-class resources
+
+There are some resources that applications are forced to share, such as cache,
+memory bandwidth, and disk I/O. There are technologies for managing those
+resources–making it possible to prioritize or isolate applications–but
```
[the remark this answers is not in the document]
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md]: Thanks, fixed

File: content/en/blog/_posts/qos-class-resources.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+  allowed values for that resource
+
+In the sections below, we discuss two types of QoS-class resources that are
+already supported by container runtimes (CRI-O and containerd).
+
+### Cache and memory bandwidth
+
+Intel® Resource Director Technology (Intel® RDT) is a technology for
+controlling the cache and memory bandwidth allocation of applications. It
+provides in hardware a class-based approach for QoS control of these shared
```
[the remark this answers is not in the document]
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md]: Thanks, changed as suggested

File: content/en/blog/_posts/qos-class-resources.md (hunk 1 above)
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md]: Thanks, changed as suggested

File: content/en/blog/_posts/qos-class-resources.md [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+  - CRI-O [v1.22][crio-release] (or later) or containerd mainline (no release
+    with blockio support is available, yet)
+  - runc [v1.0][runc-release-10] (or later)
+
+Pod annotations used for assigning the QoS-class resources:
+
+| Annotation  | Description
+| :---------- | :----------
+| `rdt.resources.beta.kubernetes.io/pod` | set a Pod-level default RDT class for all containers
+| `rdt.resources.beta.kubernetes.io/container.<container-name>` | set ‘RDT class of one container
```
[the remark this answers is not in the document]
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md]: Woops, fixed

File: content/en/blog/_posts/qos-class-resources.md [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+| `blockio.resources.beta.kubernetes.io/pod` | set a Pod-level default blockio class for all containers
+| `blockio.resources.beta.kubernetes.io/container.<container-name>` | set blockio class of one container
+
+Configuration file format is described in detail with more examples
+[here for RDT][rdt-config-doc], and [here for blockio][blockio-config-doc].
+
+### An example
+
+The following example demonstrates the configuration of RDT and blockio
+QoS-class resources with CRI-O runtime
```
[the remark this answers is not in the document]
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md]: Thanks, fixed

File: content/en/blog/_posts/qos-class-resources.md [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+blockio_config_file = "/etc/crio/blockio.conf.yaml"
+...
+```
+
+#### Assigning QoS-class resources
+
+Below is a referential example leveraging both RDT and blockio resources. It
+creates a Pod of three containers with:
+
+- all containers are assigned to the “limited” RDT class by default, except for
```
[the remark this answers is not in the document]
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md:235]: Thanks, fixed

File: content/en/blog/_posts/qos-class-resources.md [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+```
+
+Note that the containers are minimal pause images, doing nothing in practice,
+but just demonstrating how different containers are assigned to different
+classes.
+
+### Limitations of the runtime-only approach
+
+The biggest limitations in this runtime-only approach probably lies in the user
+interface. Users have no visibility what QoS-class resource types (or classes
```
[the remark this answers is not in the document]
Comment (marquiz) [reply content/en/blog/_posts/qos-class-resources.md]: Thanks, fixed

**Comment by marquiz** [comment]
> * given the target audience and early state for this feature, the article is a better fit for https://kubernetes.dev/blog/

Yeah, sure. That might actually be better. Should I just submit this to https://github.com/marquiz/contributor-site?

> * we should not publish the article text as-is

Do you have some more specific feedback on this?

**Comment by marquiz** [comment]
Submitted https://github.com/kubernetes/contributor-site/pull/332

**Comment by sftim** [comment]
> we should not publish the article text as-is

We wouldn't announce a yet-to-be-accepted KEP to end users. Usually we announce new features when they are part of a release.

**Comment by sftim** [comment]
Given that there is https://github.com/kubernetes/contributor-site/pull/332 now, I will close this.
/close

**Pull request closed without merging**